src/txn/txn.c

```diff
@@ -1669,13 +1669,38 @@ __txn_mod_compare(const void *a, const void *b)
     return (0);
 }
 
+/*
+ * __txn_check_if_stable_has_moved_ahead_commit_ts --
+ *     Check if the stable timestamp has moved ahead of the commit timestamp.
+ */
+static int
+__txn_check_if_stable_has_moved_ahead_commit_ts(WT_SESSION_IMPL *session)
```

Comment:
Overall change looks good to me, I think this separate function might not be required, as this is more specific to commit.

Reply:
@bvpvamsikrishna I prefer this to be a function for code modularization, but I am happy to change if you have a strong opinion.